tests/check_xcode.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+# Script that checks if the XCode (MacOS) project is complete.
+# If the XCode project is incomplete, then this script also uses
+# mod-pbxproj (https://github.com/kronenthaler/mod-pbxproj) to
+# suggest fixes/additions to make the XCode project complete.
+#
+# mod-pbxproj should already have been installed (through PIP) before
+# running this script.
+
+
+# Determine path of the current script
+HERE=$(cd `dirname $0` && pwd)
+
+# Go to toplevel ES directory
+cd ${HERE}
+cd ..
+
+ESTOP=$(pwd)
+XPROJECT=${ESTOP}/EndlessSky.xcodeproj
+
+echo "script-dir: ${HERE}"
+echo "es-top-dir: ${ESTOP}"
+
+NUM_ADDED=0
+for FILE in $(ls -1 source)
```

Comment:
not files in subdirectories, but it will loop over the subdirectory names itself.
a better alternative would be `find source -type f -printf "%f\n"` (`type` excludes directories, `printf` ensures only filenames are printed)

Reply:
I used a different implementation that preserves the paths under source (for files in subdirectories), since I have the impression that Xcode requires the paths to be complete (not just the filenames).

Reply:
Can you verify that somehow? E.g. open a PR to your fork, based from this branch, that creates and uses `source/foo/bar.*`, and then commit the diff that this check suggests to ensure the project builds?

Also, should we provide instructions for how one applies the generated diff?

Reply:
> Can you verify that somehow? E.g. open a PR to your fork, based from this branch, that creates and uses `source/foo/bar.*`, and then commit the diff that this check suggests to ensure the project builds?

I already did some basic verification in a Virtual Machine on my laptop (added a subdirectory called subdir with in it SubFile.cpp and SubFile.h). The SubFile files get added into the XCode project file at the locations where I expect them and the relative paths looked good.

Getting this to build on XCode would be more work, since I would also need to put some sensible content in the SubFile files and trigger the PRs and CI jobs. Let's keep that for when we actually start moving things to sub-directories.

> Also, should we provide instructions for how one applies the generated diff?

I don't know; this is not an end-user feature, but a feature for a developer that is making a non-trivial change (since he/she needs to add new files). Can we expect such developer to know (or at least be able to lookup) how to apply a diff?

If we would have hundreds of new files added every week, then we could also consider to make applying the diff very user-friendly by putting the diff in an automatically generated PR and thus change the interface to apply the diff to just a simple merge-button. But this looks quite a bit overkill to me with the current volume of incoming changes on ES.

Reply:
> Also, should we provide instructions for how one applies the generated diff?

I'd argue that a person who modifies the XCode project and presumably adds new libraries should be able to apply diffs.